cep-XXXX.md

```diff
@@ -0,0 +1,226 @@
+> NOT be used as such.
+
+#### Filenames
+
+The filename of distributable conda artifacts is obtained by adding the artifact extension to its
+distribution string. It MUST match this syntax:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+The maximum length of a filename MUST NOT exceed 211 characters.
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+### Identifying channels
+
+A conda channel is defined as a URL where one can find one or more `repodata.json` files arranged
+in one subdirectory (_subdir_) each. `noarch/repodata.json` MUST be present to consider the parent
+location a channel.
+
+#### Channel base URLs
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge`. For local
+repodata such as `file:///home/username/channel/noarch/repodata.json`, the channel base URL is
+`file:///home/username/channel`.
+
+When present, each path component MUST only contain lowercase ASCII letters, numbers, underscores,
+periods, and dashes. They MUST NOT start with a period or a dash. They SHOULD start and end with a
+letter or a number. If present, each path component MUST match this regex:
+
+```re
+^[a-z0-9_][a-z0-9_.-]*$
+```
+
+For `file://`-based channel URLs, the path component rules MAY be understood as recommendations
+only.
+
+The maximum length of an individual path component in a channel base URL MUST NOT exceed 128
+characters. The maximum length of a channel base URL SHOULD NOT exceed 256 characters.
+
+#### Channel names
+
+For convenience, the channel _name_ is defined as the concatenation of `scheme`, `authority` and
+`path` components of a channel URL. At least one of `authority` or `path` SHOULD be present. In
+their absence, the channel name MUST be considered empty, regardless the scheme. Empty channel
+names SHOULD NOT be used.
+
+When the scheme and authority fields are missing, the full URL can be inferred with these rules:
+
+- If the channel name matches the regex `^\.{0,2}[/\\].*$`, or if it matches the regex
+  `^[A-Z]:([\\/].*)?$` (for Windows drives), it SHOULD be understood as the path component of a
+  `file://` URL.
+- Otherwise, it SHOULD be understood as a `http[s]://` URL. The tool SHOULD assume a default scheme
+  and authority (e.g. `https://conda.anaconda.org`), and take the rest as a path component.
```

Comment:
How about
> Otherwise, the tool SHOULD provide a user-configurable mechanism to prepend a default scheme and authority, with provided channel name taken as the rest of the path component. At the time of this CEP's writing, most tool assume the default URL scheme and authority to be `https://conda.anaconda.org`.

I think this better captures the intent (i.e., support the `-c`/`--channel` option as they exist today), while recognizing that condarc already provides mechanisms to have the default schema not be `http[s]://` for certain channels.